### PR TITLE
Rename Kestrel metrics to be consistent with OpenTelemetry

### PR DESCRIPTION
**Pull request by JamesNK**
Applying changes from https://github.com/lmolkova/semantic-conventions/pull/1

@noahfalk @lmolkova

### CODE REVIEW

**Comment by JamesNK** [comment]
I haven't changed the `endpoint` tag yet. I'll look at it tomorrow.

**Review by lmolkova — APPROVED** [review]

**Comment by JamesNK** [comment]
Updated to replace endpoint.

File: src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelMetrics.cs [hunk 1]
```diff
@@ -287,11 +293,32 @@ private void TlsHandshakeStopCore(in ConnectionMetricsContext metricsContext, lo
...
+            tags.Add("server.socket.address", namedPipeEndPoint.ToString());
+            tags.Add("network.transport", "pipe");
+        }
+        else if (localEndpoint != null)
         {
-            // TODO: Improve getting string allocation for endpoint. Currently allocates.
-            // Considering adding a way to cache on ConnectionContext.
-            tags.Add("endpoint", localEndpoint.ToString());
+            tags.Add("server.socket.address", localEndpoint.ToString());
+            tags.Add("network.transport", localEndpoint.GetType().Name);
```
Comment (lmolkova) [inline src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelMetrics.cs:321]: maybe `localEndpoint.[AddressFamily](https://learn.microsoft.com/en-us/dotnet/api/system.net.endpoint.addressfamily?view=net-7.0#system-net-endpoint-addressfamily)` as a shorter and cross-language alternative?

File: src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs [hunk 2]
```diff
@@ -1192,7 +1192,7 @@ private void StartStream()
         }
 
         KestrelEventSource.Log.RequestQueuedStart(_currentHeadersStream, AspNetCore.Http.HttpProtocol.Http2);
-        _context.ServiceContext.Metrics.RequestQueuedStart(_metricsContext, AspNetCore.Http.HttpProtocol.Http2);
```
Comment (amcasey) [inline src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:1195]: I find this a little hard to read - it's not really clear from context that that's supposed to be an HTTP version.  Depending on the name of the corresponding parameter, using a named argument might help?  Failing that, a constant could have a name, "Http2MetricName" or something?

**Review by amcasey — COMMENTED** [review]
I assume this code is all too new for these renames to break anyone?

**Comment by JamesNK** [comment]
> I assume this code is all too new for these renames to break anyone?

Yes. These were added in earlier .NET 8 previews.

File: src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs (hunk 2 above)
Comment (JamesNK) [reply src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:1195]: I added constants

**Review by noahfalk — APPROVED** [review]

File: src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelMetrics.cs [hunk 3]
```diff
@@ -287,19 +300,92 @@ private void TlsHandshakeStopCore(in ConnectionMetricsContext metricsContext, lo
...
+    public static bool TryGetHandshakeProtocol(SslProtocols? protocols, [NotNullWhen(true)] out string? name, [NotNullWhen(true)] out string? version)
+    {
+        if (protocols != null)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+#pragma warning disable SYSLIB0039 // Type or member is obsolete
+            switch (protocols.Value)
+            {
+                case SslProtocols.Ssl2:
+                    name = "ssl";
```
Comment (lmolkova) [inline src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelMetrics.cs:359]: do we still need SSL?

https://github.com/dotnet/aspnetcore/issues/49857:

> Do we need tls.protocol.name if using any SSL version with SslStream on dotnet 8 and above is impossible?

Comment (JamesNK) [reply src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelMetrics.cs:359]: I'm leaving it here in an abundance of caution. I'll leave a comment.

**Pull request merged**